packages/web/src/js/header.js

```diff
@@ -0,0 +1,142 @@
+import SelectorEngine from './dom/selector-engine';
+import EventHandler from './dom/event-handler';
+import BaseComponent from './base-component';
+import { getElementFromSelector } from './utils/index';
+
+const HEADER_TOGGLE_SELECTOR = '[data-toggle="header"]';
+const HEADER_DISMISS_ATTRIBUTE = 'data-dismiss';
+const HEADER_BREAKPOINT = 1280;
+const OPEN_CLASSNAME = 'is-open';
+const BACKDROP_TAG_NAME = 'div';
+const BACKDROP_CLASSNAME = 'Header__backdrop';
+
+
+class Header extends BaseComponent {
+  constructor(selector) {
+    super(selector);
+
+    this.isShown = false;
+    this.backdrop = Header.initBackdrop(selector);
+  }
+
+  static initBackdrop (target) {
+    const backdropEl = document.createElement(BACKDROP_TAG_NAME);
+    backdropEl.classList.add(BACKDROP_CLASSNAME);
+
+    if (!SelectorEngine.findAll(`.${BACKDROP_CLASSNAME}`).length) {
+      target.appendChild(backdropEl);
+    }
+
+    return backdropEl;
+  }
+
+  onClick(event) {
+    if (
+      event.target.hasAttribute(HEADER_DISMISS_ATTRIBUTE) || event.target.classList.contains(BACKDROP_CLASSNAME)
+    ) {
+      event.preventDefault()
+      event.stopPropagation()
+      this.hide(event)
+    }
+  }
+
+  onWindowResize(event) {
+    if (event.target.innerWidth >= HEADER_BREAKPOINT) {
+      this.hide(event);
+    }
+  };
+
+  onKeyPress(event) {
+    // Do nothing if the event was already processed.
+    if (event.defaultPrevented) {
+      return;
+    }
+
+    if (event.key === 'Esc' || event.key === 'Escape') {
+      this.hide(event);
+    }
+
+    // @TODO: retain focus when using tab
+
+    // Cancel the default action to avoid it being handled twice.
+    event.preventDefault();
```

Comment:
⚠️ Once a dialog has been open, it's no longer possible to use any keyboard shortcuts (browser's or system-wide, including Tab) except Escape to close the dialog. In my code, I "solved" this with preventing default event only on the Escape key:

```js
if (event.key === 'Escape') {
  closeOpenHeaderPanes();

  // Cancel the default action to avoid it being handled twice.
  event.preventDefault();
}
```

But I'm not sure this is the proper solution.